tuf/repository_tool.py

```diff
@@ -1814,12 +1880,18 @@ def add_delegated_role(self, rolename, targets_object):
       raise securesystemslib.exceptions.FormatError(repr(targets_object) + ' is'
           ' not a Targets object.')
 
-
-    if rolename in self._delegated_roles:
-      logger.debug(repr(rolename) + ' already exists.')
+    if succinct:
+      if rolename in self._succinct_delegations:
+        logger.debug(repr(rolename) + ' already exists.')
+      else:
+        self._succinct_delegations[rolename] = targets_object
 
     else:
-      self._delegated_roles[rolename] = targets_object
```

Comment:
I'd just leave this as-is and always assign. The debug info isn't useful, I don't think.